### PR TITLE
[bugfix] Fix parsing +00:00 in RFC2822, fixes #3920

### LINKED ISSUE #3920: Parsing `+0000` timezones in RFC 2822 strings incorrectly
When parsing a date with a `+0000` timezone modifier at the end (RFC format), it parses it incorrectly. 

    $> moment.version
    '2.18.1'
    $> moment('Mon, 02 Jan 2017 06:00:00 -0800').format()
    '2017-01-02T01:00:00-05:00'
    $> moment(new Date('Mon, 02 Jan 2017 06:00:00 -0800')).format()
    '2017-01-02T09:00:00-05:00'

I am currently in EST, so the second output is what the actual time should be. 

**Environment:**
Chrome 57 and NodeJS 7.x on Windows 10

```js
console.log( (new Date()).toString()) // Wed Apr 19 2017 19:58:19 GMT-0400 (Eastern Daylight Time)
console.log((new Date()).toLocaleString()) // 4/19/2017, 7:58:19 PM
console.log( (new Date()).getTimezoneOffset())  // 240
console.log( navigator.userAgent) // ReferenceError: navigator is not defined
console.log(moment.version) // 2.18.1
```

### PR DESCRIPTION
**Pull request by TracyGJG**
Separated change - Fix for issue #3920. Includes changes made on #3935.

### CODE REVIEW

**Comment by icambron** [comment]
Fix looks good to me

File: src/test/moment/create.js [hunk 1]
```diff
@@ -475,17 +475,26 @@ test('parsing RFC 2822', function (assert) {
     }
 });
 
+test('RFC 2822 offset [issue #3920]', function (assert) {
+    var timezoneTestString = 'Mon, 02 Jan 2017 06:00:00 -0800';
+    var timezoneTestDate = new Date(timezoneTestString);
```
Comment (maggiepint) [inline src/test/moment/create.js]: There are no guarantees in the ECMAScript spec that this kind of parse will work. Are we comfortable with this test? It *may* work, but there are no guarantees. Removing release label for now.

Comment (ichernev) [inline src/test/moment/create.js]: Can we just use moment with a full format to parse this?

Comment (icambron) [inline src/test/moment/create.js]: In practice, seems like environments always let you parse their own toString() format

**Comment by ichernev** [comment]
According to this excerpt from rfc2822, the military zones should all be disregarded (-0000 means TZ is missing/not provided properly), including `Z`.

```
   The 1 character military time zones were defined in a non-standard
   way in [RFC822] and are therefore unpredictable in their meaning.
   The original definitions of the military zones "A" through "I" are
   equivalent to "+0100" through "+0900" respectively; "K", "L", and "M"
   are equivalent to  "+1000", "+1100", and "+1200" respectively; "N"
   through "Y" are equivalent to "-0100" through "-1200" respectively;
   and "Z" is equivalent to "+0000".  However, because of the error in
   [RFC822], they SHOULD all be considered equivalent to "-0000" unless
   there is out-of-band information confirming their meaning.
```

**Comment by TracyGJG** [comment]
Pull Request withdrawn.

**Comment by ichernev** [comment]
You should not withdraw it because of the military stuff, but because of the rewrite. I was just a bit stunned that the old code was wrong because of that, too.

**Comment by icambron** [comment]
Yeah, that's why it wasn't in the Luxon code that got back ported into the rewrite PR. Not only is it very rarely used, it's been explicitly cast aside. I thought I had pasted that into one of these PRs, but I've lost track of all our 2822 issues/pulls.

**Pull request closed without merging**